### ISSUE TITLE
Feature Request: Support for git dependency with subdirectory path specified

### ISSUE THREAD

**Issue by harryeakins** [CocoaPods/CocoaPods#9805]
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

I use a monorepo for development. I have a CocoaPod inside the repo at `/packages/my_package/my_package.podspec`

Another team has a project which wants to depend on this package. They enter this into their Podfile:
```
pod 'MyPackage', :git => 'https://github.com/harryeakins/monorepo.git', :path => './packages/my_package'
```

When they run `pod install`, this happens:
```
[!] The dependency `MyPackage` specifies `podspec` or `path` in combination with other download strategies. This is not allowed.
```

## What did you expect to happen?

It should fetch the git repository, and use the CocoaPod from the correct directory provided. 

I *think* this is pretty reasonable. It would work in some similar package managers, such as  Dart/Flutter: https://dart.dev/tools/pub/dependencies

## What happened instead?

It showed an error.

After further investigation, I realized I can put multiple podspec files in the root directory of the repository. This is not ideal though... it would be nice to keep our repository neat and tidy with the podspec files belonging to the packages they describe.

### Stack

```
   CocoaPods : 1.8.4
        Ruby : ruby 2.6.3p62 (2019-04-16 revision 67580) [universal.x86_64-darwin19]
    RubyGems : 3.0.3
        Host : Mac OS X 10.15.4 (19E287)
       Xcode : 11.4 (11E146)
         Git : git version 2.24.1 (Apple Git-126)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib
Repositories : trunk - CDN - https://cdn.cocoapods.org/
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-binary       : 0.4.4
cocoapods-deintegrate  : 1.0.4
cocoapods-dependencies : 1.3.0
cocoapods-plugins      : 1.0.0
cocoapods-search       : 1.0.0
cocoapods-stats        : 1.1.0
cocoapods-trunk        : 1.4.1
cocoapods-try          : 1.1.0
```

**Comment by alexlee002** [comment]
I don't think is a good way to share your pods.

can the other team has the permission change your code? if no, why not use a released binary pod instead?

if so, why not make a separate code repo for the share pod?

**Comment by harryeakins** [comment]
> I don't think is a good way to share your pods.

Why not?

> can the other team has the permission change your code? if no, why not use a released binary pod instead?

We want the other teams to be able to read our code and submit pull requests against it, so I don’t think this is an option.

> if so, why not make a separate code repo for the share pod?

We’ve decided to use a monorepo (https://en.m.wikipedia.org/wiki/Monorepo) because the pros outweigh the cons for our development process. Asking us to move away from this model  is asking us to sacrifice development productivity. It would be great to support different development practices.

**Comment by orta** [comment]
This seems reasonable, but it should shouldn't be using `:path` - perhaps `:dir` or `:root`? 

It's less accurate, but `:path` means that it will use the local fs to find a pod and make a dev pod in CocoaPods. Allowing the option to act differently depending on `git:` is gonna be confusing for docs and readability

**Comment by alexlee002** [comment]
@harryeakins I read the monorepo before.  But I don't think software development is something like Religion.  We can use the "right way" to us.

But there is a creed that I think is right: "Keep it simple".  Cocoapod is now becoming more complex. I think it's better to build a ecology with much plugins. so, I think you can write you own pod-plugin to customize your pod.

BTW, actually,  my team currently use the dev-pattern much like monorepo,   there are also some "base feature kits" need to share with other teams,  and I separate them into isolate repos, so, actually I'm using the pattern mixed with "monorepo" and "multi-repos".  I wrote my own pod-plugin to manage this repos and make them looks like a single repo. it works good.

**Comment by alexlee002** [comment]
@orta I think it will bring something hard to control.  

For example,  there is a big monorepo project (eg, named "MyProj".) has   pod-a, pod-b, pod-c, ...
And， my new project needs pod-a in branch master, and pod-b in tag-v1.0,  and pod-c in commit 666aaaa,  you need to clone 3-times of this big repo?

**Comment by orta** [comment]
If you're operating internally in a monorepo they should be using `:path` to refer to the other Pods given that the source is right there (or make a plugin like you have). Having support for `:git` with :`dir` (or whatever) allows you to not need the Pod to be the root of your repo and is globally useful to any git repo,

**Comment by alexlee002** [comment]
@orta but it is not easy to solve my problem: if the monorepo is a huge repo, and other project need to refer some of the pods in this monorepo but they are in different branches or tags or commits,   how to avoid clone many times of this huge repo?

**Comment by harryeakins** [comment]
It's worth noting that this issue will potentially go away when git sparse-checkout becomes more mature and supported: https://github.blog/2020-01-17-bring-your-monorepo-down-to-size-with-sparse-checkout/

**Comment by alexlee002** [comment]
@harryeakins

No, I am not fully agree.  Even if we use sparse-checkout, we still need to download the huge `.git` objects one or more times（for example,  my git repo is more than 20G -- a long history project, and some bad guy pushed some big binary files before）,  and sparse-checkout is still toooo slow.
Maybe we can clone the repo once,  and copy the first pod-dir,  and then checkout to next pod's branch/commit, and copy the next pod-dir?

**Comment by chj-damon** [comment]
@harryeakins what is the final solution to this issue? I'm facing the same problem as you.

**Comment by ymaniz09** [comment]
I'm facing the same issue as well

**Comment by alfergus0n** [comment]
I assume this has never been addressed? With the rise of monorepos, it would make sense to accommodate for such. Maybe something like `:package`

**Comment by alexlee002** [comment]
Another question but little relative:  Which is the best way to use a monorepo with multiple APPs?

**Comment by alexlee002** [comment]
I think this should extension the `source` property in `podspec`,  e.g.:

  spec.source = {:git => 'ssh://host/pod.git', :pod_dir => 'path/to/pod' }
